lib/src/edhoc.rs

```diff
@@ -638,17 +641,18 @@ fn edhoc_kdf(
 }
 
 fn encode_plaintext_3(
-    id_cred_i: &BytesIdCred,
+    id_cred_i: &IdCred,
     mac_3: &BytesMac3,
     ead_3: &Option<EADItem>,
 ) -> Result<BufferPlaintext3, EDHOCError> {
     let mut plaintext_3: BufferPlaintext3 = BufferPlaintext3::new();
 
     // plaintext: P = ( ? PAD, ID_CRED_I / bstr / int, Signature_or_MAC_3, ? EAD_3 )
-    plaintext_3.content[0] = id_cred_i[id_cred_i.len() - 1]; // hack: take the last byte of ID_CRED_I as KID
-    plaintext_3.content[1] = CBOR_MAJOR_BYTE_STRING | MAC_LENGTH_3 as u8;
-    plaintext_3.content[2..2 + mac_3.len()].copy_from_slice(&mac_3[..]);
-    plaintext_3.len = 2 + mac_3.len();
+    id_cred_i.write(&mut plaintext_3)?;
```

Comment:
This gives the impression that `id_cred_i` is being written on. Would something like the code below work?

```
plaintext_3.extend_from_slice(id_cred_i.as_cbor())
```

Or else just `s/write/write_to_message/`